src/test/ui/raw-ref-op/raw-ref-temp.rs

```diff
@@ -0,0 +1,34 @@
+// Ensure that we don't allow taking the address of temporary values
+#![feature(raw_ref_op)]
+
+const PAIR: (i32, i64) = (1, 2);
+const PAIR_REF: &(i32, i64) = &(1, 2);
+
+const ARRAY: [i32; 2] = [1, 2];
+const ARRAY_REF: &[i32; 2] = &[3, 4];
+const SLICE_REF: &[i32] = &[5, 6];
+
+fn main() {
+    let ref_expr = &raw const 2;                    //~ ERROR cannot take address
+    let mut_ref_expr = &raw mut 3;                  //~ ERROR cannot take address
+    let ref_const = &raw const 4;                   //~ ERROR cannot take address
+    let mut_ref_const = &raw mut 5;                 //~ ERROR cannot take address
+
+    let field_ref_expr = &raw const (1, 2).0;       //~ ERROR cannot take address
+    let mut_field_ref_expr = &raw mut (1, 2).0;     //~ ERROR cannot take address
+    let field_ref = &raw const PAIR.0;              //~ ERROR cannot take address
+    let mut_field_ref = &raw mut PAIR.0;            //~ ERROR cannot take address
+
+    let field_ref_expr = &raw const [1, 2][0];       //~ ERROR cannot take address
+    let mut_field_ref_expr = &raw mut [1, 2][0];     //~ ERROR cannot take address
+    let field_ref = &raw const ARRAY[0];             //~ ERROR cannot take address
+    let mut_field_ref = &raw mut ARRAY[1];           //~ ERROR cannot take address
+
+    // These are all OK, we're not taking the address of the temporary
```

Comment:
General note: could we move more of the "OK" things in various tests added in this PR to `// check-pass` files so as to have a better split? (That makes it easier for automated tools and to have a better set of specifying tests for the language.) (I'm thinking of this test and the borrowck ones.)